apps/site/lib/site_web/views/trip_plan_view.ex

```diff
@@ -589,4 +601,22 @@ defmodule SiteWeb.TripPlanView do
       }
     )
   end
+
+  @spec get_highest_one_way_fare(TripPlan.Itinerary.t()) :: integer
+  def get_highest_one_way_fare(itinerary) do
+    itinerary.legs
+    |> Enum.filter(fn leg -> leg.description != "WALK" end)
+    |> Enum.reduce(0, fn leg, acc ->
+      highest_one_way_fare =
+        leg
+        |> Kernel.get_in([
+          Access.key(:mode, %{}),
+          Access.key(:fares, %{}),
+          Access.key(:highest_one_way_fare, %{}),
+          Access.key(:cents, 0)
```

Comment:
TIL about `Access`,  I'm guessing you need these to ensure the lookup never returns `nil`? Very cool!

Reply:
Yes, I've seen it used instead of the dot notation to access deeply nested structures and since it has a default field, we make sure nothing breaks.

Reply:
That's cool, I didn't know this technique either. Another option that's similar is `Map.get/3` which takes an optional default value instead of the `nil` default.